code/game/objects/items/melee/misc.dm

```diff
@@ -74,11 +74,37 @@
 	custom_materials = list(/datum/material/iron = 1000)
 	wound_bonus = 10
 	bare_wound_bonus = 25
+	var/lunge_delay = 1.2 SECONDS
 
 /obj/item/melee/sabre/Initialize()
 	. = ..()
 	AddComponent(/datum/component/butchering, 30, 95, 5) //fast and effective, but as a sword, it might damage the results.
 
+/obj/item/melee/sabre/examine(datum/source, mob/user, list/examine_list)
+	. = ..()
+	. += "<span class='notice'><b>right-click</b> to lunge with [src].</span>"
+
+/obj/item/melee/sabre/afterattack_secondary(atom/target, mob/user, proximity_flag, click_parameters)
+	if(user.Adjacent(target))
+		return
+	to_chat(user, "<span class='notice'>You prepare to lunge at [target]!</span>")
+	. = SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	if(!do_after(user, lunge_delay, interaction_key = "lunge_attack"))
+		return
+	user.throw_at(target, 2, 10, spin = FALSE, gentle = TRUE, callback = CALLBACK(src, .proc/after_lunge, user, target))
```

Comment:
I don't think there's a better alternative, but this is a bit silly

Reply:
The bigger issue is that lying down makes you sail clean over the target since it has such absurd lunge range. Since there is no check for distance between you and the target, this overcompensation really just ends up being it's downfall more than a benefit. If it was the distance between target and user +2, maybe then it would be slightly more reasonable.